src/Features/VisualBasic/Portable/CodeRefactorings/SyncNamespace/VisualBasicSyncNamespaceService.vb

```diff
@@ -0,0 +1,79 @@
+        Inherits AbstractSyncNamespaceService(Of NamespaceStatementSyntax, CompilationUnitSyntax)
+
+        Public Overrides Function TryGetReplacementReferenceSyntax(reference As SyntaxNode, newNamespaceParts As ImmutableArray(Of String), ByRef old As SyntaxNode, ByRef [new] As SyntaxNode) As Boolean
+            Dim nameRef = TryCast(reference, SimpleNameSyntax)
+            If nameRef IsNot Nothing Then
+                Dim outerMostNode = GetQualifiedNameSyntax(nameRef)
+                old = outerMostNode
+
+                If outerMostNode Is nameRef Or newNamespaceParts.IsDefaultOrEmpty Then
+                    [new] = outerMostNode
+                Else
+                    If newNamespaceParts.Length = 1 And newNamespaceParts(0).Length = 0 Then
+                        [new] = nameRef
+                    Else
+                        Dim qualifiedNamespaceName = CreateNameSyntax(newNamespaceParts, newNamespaceParts.Length - 1)
+                        [new] = SyntaxFactory.QualifiedName(qualifiedNamespaceName, nameRef)
+                    End If
+                    [new] = [new].WithTriviaFrom(outerMostNode)
+                End If
+                Return True
+            Else
+                old = Nothing
+                [new] = Nothing
+                Return False
+            End If
+        End Function
+
+        Protected Overrides Function EscapeIdentifier(identifier As String) As String
+            Return identifier?.EscapeIdentifier()
+        End Function
+
+        ' This is only reachable when called from a VB refacoring provider, which is not implemented yet.
+        Protected Overrides Function ChangeNamespaceDeclaration(root As SyntaxNode, declaredNamespaceParts As ImmutableArray(Of String), targetNamespaceParts As ImmutableArray(Of String)) As SyntaxNode
+            Throw ExceptionUtilities.Unreachable
+        End Function
+
+        ' This is only reachable when called from a VB refacoring provider, which is not implemented yet.
+        Protected Overrides Function GetMemberDeclarationsInContainer(compilationUnitOrNamespaceDecl As SyntaxNode) As IReadOnlyList(Of SyntaxNode)
+            Throw ExceptionUtilities.Unreachable
+        End Function
+
+        ' This is only reachable when called from a VB refacoring provider, which is not implemented yet.
+        Protected Overrides Function ShouldPositionTriggerRefactoringAsync(document As Document, position As Integer, cancellationToken As CancellationToken) As Task(Of (Boolean, NamespaceStatementSyntax))
+            Throw ExceptionUtilities.Unreachable
+        End Function
+
+        Private Function GetQualifiedNameSyntax(node As NameSyntax) As NameSyntax
+            Dim parent = TryCast(node.Parent, QualifiedNameSyntax)
+            If parent IsNot Nothing Then
+                Return If(parent.Right Is node, parent, node)
+            Else
+                Return node
+            End If
+        End Function
```

Comment:
i don't know why the VB code seems to be repeating a bunch of hte logic in the C# code.

Reply:
noted. Will clean up later.